jboss-eap-standalone-rhel7/azuredeploy.json

```diff
@@ -0,0 +1,285 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {    
+    "adminUsername": {
+      "type": "string",
+      "metadata": {
+        "description": "Linux VM user account name"
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Administrator password for Linux VM user account"
+      }
+    },
+    "dnsLabelPrefix": {
+      "type": "string",
+      "metadata": {
+        "description": "DNS Label for the Public IP. Must be lowercase. It should match with the following regular expression: ^[a-z][a-z0-9-]{1,61}[a-z0-9]$ or it will raise an error."
+      }
+    },
+    "eapUserName": {
+      "type": "string",
+      "metadata": {
+        "description": "User name for EAP manager"
+      }
+    },
+    "eapPassword": {
+      "type": "string",
+      "metadata": {
+        "description": "Password for EAP manager"
+      }
+    },
+
+    "rhsmUserName": {
+      "type": "string",
+      "defaultValue": "mfentane@redhat.com",
+      "metadata": {
+        "description": "User name for Red Hat subscription manager"
+      }
+    },
+    "rhsmPassword": {
+      "type": "securestring",
+      "defaultValue": "Myr1am84",
+      "metadata": {
+        "description": "Password for Red Hat subscription  manager"
+      }
+    },
+    "eapProfile": {
+      "type": "string",
+      "metadata": {
+        "description": "EAP profile to use"
+      }
+    },
+    "sshPassPhrase": {
+      "type": "string",
+      "metadata": {
+        "description": "Pass phrase for SSH certificate"
+      }
+    }
+  },
+  "variables": {
+    "imagePublisher": "redhat",
+    "imageOffer": "rhel",
+    "imageSKU":"7.2",
+    "baseName": "[uniquestring(parameters('dnsLabelPrefix'),resourceGroup().id,uniquestring(deployment().name))]",
```

Comment:
remove deployment name since it's not guaranteed to be idempotent

Reply:
done!

Reply:
done!